language-plutus-core/examples/Language/PlutusCore/Examples/Data/InterList.hs

```diff
@@ -39,8 +38,8 @@ We encode the following in this module:
 --
 -- > fix \(interlist :: * -> * -> *) (a :: *) (b :: *) ->
 -- >     all (r :: *). r -> (a -> b -> interlist b a -> r) -> r
-getBuiltinInterList :: Quote (RecursiveType ())
-getBuiltinInterList = do
+interListData :: RecursiveType ()
+interListData = runQuote $ do
```

Comment:
There isn't much point in escaping `Quote` here, but I also think there isn't much danger, because we return a type (in this situation it's allowed to escape `Quote`) and a function, i.e. a Haskell value (even though it's a function from `Term` to `Term`, which can be potentially dangerous, but then it already was dangerous).

In general, we should avoid reasoning about where it's safe and where it's not safe to escape `Quote`, but I think it's fine to have some corner cases that we thought about seriously.